packages/grid/x-grid-modules/src/components/column-header-item.tsx

```diff
@@ -55,6 +56,7 @@ export const ColumnHeaderItem = React.memo(
           maxWidth: width,
           maxHeight: headerHeight,
           minHeight: headerHeight,
+          borderRight: showColumnRightBorder ? '1px solid #bdc3c7' : 'none',
```

Comment:
I think that we should use a CSS class name. Inline-style should only be our last resort for performance considerations.